flink-state-backends/flink-statebackend-common/src/main/java/org/apache/flink/state/common/ChangelogMaterializationMetricGroup.java

```diff
@@ -51,14 +57,17 @@ public ChangelogMaterializationMetricGroup(MetricGroup parentMetricGroup) {
                 counter(COMPLETED_MATERIALIZATION, new ThreadSafeSimpleCounter());
         this.failedMaterializationCounter =
                 counter(FAILED_MATERIALIZATION, new ThreadSafeSimpleCounter());
+
+        gauge(LAST_DURATION_OF_MATERIALIZATION, () -> lastDuration);
     }
 
     void reportStartedMaterialization() {
         startedMaterializationCounter.inc();
     }
 
-    void reportCompletedMaterialization() {
+    void reportCompletedMaterialization(long duration) {
```

Comment:
How about marking `triggerTime` as a field of this class?  This way we don't have to pass `long duration` in `reportCompletedMaterialization`.

Like `AbstractCheckpointStats#getEndToEndDuration`.

Reply:
Hi @fredia , I prefer to pass in duration directly by the caller, so that ChangelogMaterializationMetricGroup can not depend on the current time.
AbstractCheckpointStats#getEndToEndDuration uses last_ack_time - trigger_time because the checkpoint may not be completed and the duration may change.

Reply:
I see, thanks for your clarification. 👍